sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -828,6 +828,8 @@ private[hive] class HiveClientImpl(
       }
     }
 
+    // Hive query needs to start SessionState.
+    SessionState.start(state)
```

Comment:
@HyukjinKwon @wangyum should this also be replaced with `shim.setCurrentSessionState(state)`?

Reply:
I think it's because it executes some Hive queries e.g.) `sparkSession.metadataHive.runSqlHive("SELECT * FROM t")` at `sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreCatalogSuite.scala`

Reply:
Yes. Hive query needs to start `SessionState`.